core/src/main/scala/org/apache/spark/h2o/backends/internal/H2ORpcEndpoint.scala

```diff
@@ -48,12 +49,19 @@ class H2ORpcEndpoint(override val rpcEnv: RpcEnv) extends ThreadSafeRpcEndpoint
 
     case StopEndpointMsg =>
       this.stop()
+    case LockClusterMsg() =>
+      Paxos.lockCloud("Locking the cloud from Sparkling Water as we have reached the expected cluster size.")
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
     case StartH2OWorkersMsg(conf) =>
       val nodeDesc = InternalH2OBackend.startH2OWorker(conf)
       context.reply(nodeDesc)
+    case CheckClusterSizeMsg() =>
+      context.reply(H2O.CLOUD.size())
+    case IsLeaderNodeMsg() =>
+      val currentNode = NodeDesc(SparkEnv.get.executorId, H2O.SELF_ADDRESS.getHostAddress, H2O.API_PORT)
+      context.reply((currentNode, H2O.SELF.isLeaderNode))
```

Comment:
Do you think is possible to add a flag of being a leader node to directly to `NodeDesc`?

Reply:
I have changed this method to co return Option[NodeDesc] we so don't have to send two items. I would prefer to keep the NodeDesc as it is as the information about leader is used only here but NodeDesc is used at a lot of places around the codebase